pkg/namespace/namespace.go

```diff
@@ -0,0 +1,99 @@
+package namespace
+
+import (
+	"strings"
+
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+// Valid validates all linux namespace mode excepts network mode.
+func Valid(ns specs.LinuxNamespaceType, mode string) bool {
+	parts := strings.SplitN(string(mode), ":", 2)
+
+	switch ns {
+	case specs.UserNamespace:
+		switch v := parts[0]; v {
+		case "", "host":
+		default:
+			return false
+		}
+
+	case specs.PIDNamespace, specs.UTSNamespace:
+		switch v := parts[0]; v {
+		case "", "host":
+		case "container":
+			if len(parts) != 2 || parts[1] == "" {
+				return false
+			}
+		default:
+			return false
+		}
+
+	case specs.IPCNamespace:
+		return IsEmpty(mode) || IsNone(mode) || IsPrivate(ns, mode) || IsHost(mode) || IsShareable(mode) || IsContainer(mode)
+	}
+	return true
+}
+
+// IsEmpty indicates whether namespace mode is empty.
```

Comment:
These functions aren't namespace's functions, so I don't agree the package named `namespace`.  These just check the container's network mode.